LIBTD-1316: Import script to ingest (SEAMUS) records into COMPEL

shabububu opened the pull request:
There are 4 tasks to parse and import records into COMPEL . 

Two rake tasks just demonstrate how to parse and scrape information --

* Extract SEAMUS XML - Authors. To run: 
  `bin/rake seamus:extract_authors["input.xml","output.json"]'`
  Note that this now creates a secondary `output.json.scraped_profiles` file with information scraped from SEAMUS profiles
* Extract SEAMUS XML - Items . To run: 
  `bin/rake seamus:extract_items["input.xml","output.json"]`
  **This shouldn't have been modified for this PR, but I'm just mentioning it.**

  [Edit: For the `seamus:extract_*` tasks, if there's a space between the input and output files, the script errors out. I'm not sure why that happens, but I'm mentioning it for awareness purposes. Please let me know if there's a better way to take arguments in from rake tasks.]

The other two rake tasks parse/scrape and import data into the application --
  
* Import SEAMUS XML - Authors. To run: 
  `bin/rake seamus:import_authors["input.xml"]`
* Import SEAMUS XML - Items. Must be run AFTER import_authors. To run: 
  `bin/rake seamus:import_items["input.xml"]`

shabububu commented:
Also, I kept some of the html tags `<p>`, `<a href...`, etc in the description because I wasn't sure how to handle it. We can decide to remove them, but doing so would change the type of content the authors originally intended to put in, especially in the case of links outbound. On the flip side, keeping the html in and rendering it might make things more of a headache in terms of security.

tingtingjh commented:
@shabububu From my previous experience with IAWA stakeholders, they would like to render the HTML safe version of the description text, i.e., interpret the format tags properly and links clickable. So I am leaning toward interpreting the tags. But of course, we need to gather more requirements to be certain.

tingtingjh commented on lib/tasks/vtul/seamus.rake:
```diff
@@ -19,20 +20,55 @@ namespace :seamus do
       open(input_xml_file) do |s| content = s.read end
```
content = File.read(input_xml_file) might be cleaner?

shabububu replied:
That's true. The `do` block is probably better to use in the case where we're doing more than just reading it in. I'll try changing it.

tingtingjh commented on lib/tasks/vtul/seamus_importer.rb:
```diff
@@ -0,0 +1,221 @@
+      user.facebook_handle = handle(user_profile.member_facebook_link)
+      user.googleplus_handle = handle(user_profile.member_google_link)
+        
+      if !( user_profile.member_website_link.nil? || user_profile.member_website_link.blank?)
```
Does blank? already include nil?

shabububu replied:
Ah, you're right. I was confusing `blank?` with `empty?`.
```
irb(main):001:0> x = nil
=> nil
irb(main):002:0> x.empty?
NoMethodError: undefined method `empty?' for nil:NilClass
	from (irb):2
irb(main):003:0> x.blank?
=> true
```
I can fix it.

tingtingjh commented on lib/tasks/vtul/seamus_importer.rb:
```diff
@@ -0,0 +1,221 @@
+      performance.contributor = composition.creator
+      performance.subject = item.metadata.instrumentation
+      performance.subject = ["(none)"] if performance.subject.empty?
+      performance.description = [item.body, item.metadata.subtitle]
```
Take a look at: https://github.com/VTUL/hyrax2/blob/0fde5096eaa76ece50d610a62957d4b7bc7f52bc/app/helpers/iawa_helper.rb

shabububu commented:
I've made some modifications:
* Added in a `related_url` to hold performance related links from the SEAMUS import. 
* Added some `.html_safe` code based on IAWA to render html formatting in the work descriptions. (NOTE: Created a separate Jira ticket to assess the need for security checks when allowing users to enter html code, LIBTD-1400) 
* Modified the rake task to NOT import the SEAMUS default user profile image.
* Code cleanup based on PR feedback.

shabububu commented:
Sorry -- I added one more minor label change for the `related_url`
